server/chat-plugins/wifi.tsx

```diff
@@ -280,22 +288,40 @@ class Giveaway extends Rooms.RoomGame {
 		return set;
 	}
 
-	generateWindow(rightSide: string) {
-		let buf = `<center><h3>It's ${this.game} giveaway time!</h3>`;
-		buf += Utils.html`<small>Giveaway started by ${this.host.name}</small>`;
-		buf += `<table style="margin-left:auto;margin-right:auto">`;
-		buf += Utils.html`<tr><td colspan="2" style="text-align:center"><strong>Giver: </strong>${this.giver.name}<br /><strong>OT:</strong> ${this.ot}, <strong>TID:</strong> ${this.tid}</td></tr>`;
-		buf += `<tr><td style="text-align:center;width:45%"><psicon item="${this.ball}" /> ${this.sprite} <psicon item="${this.ball}" />`;
+	generateWindow(rightSide: Chat.VNode | string): Chat.VNode {
 		const set = Giveaway.convertIVs(this.prize, this.ivs);
-		buf += `<p>${Chat.formatText(set, true)}</p></td>`;
-		buf += `<td style="text-align:center;width:45%">${rightSide}</td></tr>`;
-		if (this.extraInfo?.trim().length) {
-			this.extraInfo = this.extraInfo.trim().replace(/<br \/>/g, '\n');
-			buf += `<tr><td colspan="2" style="text-align:center;"><strong>Extra Information</strong><br />${Chat.formatText(this.extraInfo, true)}</td></tr>`;
-		}
-		buf += `</table>`;
-		buf += `<p style="text-align:center;font-size:7pt;font-weight:bold;"><u>Note:</u> You must have a Switch, Pok&eacute;mon ${gameName[this.game]}, and Nintendo Switch Online to receive the prize. Do not join if you are currently unable to trade. Do not enter if you have already won this exact Pok&eacute;mon, unless it is explicitly allowed.</p>`;
-		return buf;
+		return <center>
+			<h3>It's {this.game} giveaway time!</h3>
+			<small>Giveaway started by {this.host.name}</small>
+			<table style={{marginLeft: 'auto', marginRight: 'auto'}}>
+				<tr>
+					<td colSpan={2} style={{textAlign: 'center'}}>
+						<strong>Giver:</strong> {this.giver.name}<br />
+						<strong>OT:</strong> {this.ot}, <strong>TID:</strong> {this.tid}
+					</td>
+				</tr>
+				<tr>
+					<td style={{textAlign: 'center', width: '45%'}}>
+						<psicon item={this.ball} /> {this.sprite} <psicon item={this.ball} />
+						<Chat.JSX.FormatText isTrusted>{set}</Chat.JSX.FormatText>
+					</td>
+					<td style={{textAlign: 'center', width: '45%'}}>{rightSide}</td>
+				</tr>
+				{this.extraInfo?.trim().length ?
+					<tr>
+						<td colSpan={2} style={{textAlign: 'center'}}>
+							<strong>Extra Information</strong><br />
+							<Chat.JSX.FormatText isTrusted>{this.extraInfo.trim().replace(/<br \/>/g, '\n')}</Chat.JSX.FormatText>
+						</td>
+					</tr> : ''}
```

Comment:
```suggestion
				{!!this.extraInfo?.trim().length && <tr>
					<td colSpan={2} style={{textAlign: 'center'}}>
						<strong>Extra Information</strong><br />
						<Chat.JSX.FormatText isTrusted>{this.extraInfo.trim().replace(/<br \/>/g, '\n')}</Chat.JSX.FormatText>
					</td>
				</tr>}
```

`&&` is the idiomatic way to do a conditional in JSX.